### PR TITLE
fileByPath added in Graphql

### PR DESCRIPTION
**Pull request by maasencioh**
* add `fileByPath` to GraphQL
* fixes relative paths
* document addon
* make relation between revisions and file's database

### CODE REVIEW

File: addons/file-sync/Resolvers/FileContent.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import { GqlResolvers } from 'App/graphql';
+
+const resolvers: GqlResolvers = {
+  Query: {
+    async fileByPath(_, { path }) {
+      const splitedPath = path.split(sep);
+      const fileName = splitedPath[splitedPath.length - 1];
+      const fileBuffer = await readFile(path);
+      const content = fileBuffer.toString();
```
Comment (targos) [inline addons/file-sync/Resolvers/FileContent.ts]: we can do it like this temporarily but this won't work for binary files. we should aim for only returning a URL in the GraphQL API and serve the file separately (with an Adonis route)

File: addons/file-sync/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Configuration
+
+Please be sure to have run previously the installation and configuration on the [backend](../../README.md#installation) and the [frontend](../../frontend/README.md#installation).
+
+After is required to have created an user:
+
+```shell
+node ace user:create
+```
```
Comment (targos) [inline addons/file-sync/README.md:13]: this should be documented in the backend readme, I think

File: addons/file-sync/Resolvers/FileContent.ts (hunk 1 above)
Comment (maasencioh) [reply addons/file-sync/Resolvers/FileContent.ts]: I tried with signedUrl from datadrive, but I have the following error
```
"extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "MethodNotSupported: E_METHOD_NOT_SUPPORTED: Method getSignedUrl is not supported for the driver LocalFileSystemStorage",
            "    at LocalFileSystemStorage.getSignedUrl (/Users/macbook/github/mylims/mylims/node_modules/@slynova/flydrive/src/Storage.ts:94:9)",
            "    at DataDrive.getSignedUrl (/Users/macbook/github/mylims/mylims/node_modules/adonis-datadrive/src/DataDrive.ts:78:36)",
            "    at fileByPath (/Users/macbook/github/mylims/mylims/addons/file-sync/Resolvers/FileContent.ts:15:56)",
            "    at processTicksAndRejections (node:internal/process/task_queues:93:5)"
          ]
        }
```

Comment (targos) [reply addons/file-sync/Resolvers/FileContent.ts]: I wasn't thinking about using a signed URL

Comment (maasencioh) [reply addons/file-sync/Resolvers/FileContent.ts]: > we should aim for only returning a URL in the GraphQL API and serve the file separately (with an Adonis route)

Sorry, what did you mean with this then?

Comment (targos) [reply addons/file-sync/Resolvers/FileContent.ts]: Just a normal URL protected with the session, like any API endpoint

**Pull request merged**